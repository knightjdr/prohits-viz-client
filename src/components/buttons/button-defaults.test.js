import React from 'react';
import { render } from '@testing-library/react';

import Button from './button-defaults';

const theme = {
  fontStackSystem: 'arial',
  timingFunction: 'ease-in',
};

describe('Default button style', () => {
  describe('default', () => {
    let container;

    beforeAll(() => {
      ({ container } = render(<Button theme={theme} />));
    });

    it('should match snapshot', () => {
      expect(container).toMatchSnapshot();
    });

    it('should set transition timing function', () => {
      expect(container.firstChild).toHaveStyleRule('transition-timing-function', theme.timingFunction);
    });

    it('should set animation timing function on active', () => {
      const modifier = {
        modifier: '&:active',
      };
      expect(container.firstChild).toHaveStyleRule('animation-timing-function', theme.timingFunction, modifier);
    });
  });

  describe('default with optional props', () => {
    let container;

    beforeAll(() => {
      ({ container } = render(
        <Button
          disabled
          shadow
          theme={theme}
        />,
      ));
    });

    it('should match snapshot', () => {
      expect(container).toMatchSnapshot();
    });

    it('should set box-shadow', () => {
      expect(container.firstChild).toHaveStyleRule('box-shadow', '0 2px 5px 0 rgba(0,0,0,0.26)');
    });

    it('should set cursor', () => {
      expect(container.firstChild).toHaveStyleRule('cursor', 'not-allowed');
    });

    it('should set opacity', () => {
      expect(container.firstChild).toHaveStyleRule('opacity', '0.6');
    });
  });
});
