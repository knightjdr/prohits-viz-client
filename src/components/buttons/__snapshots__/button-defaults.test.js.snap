// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default button style default should match snapshot 1`] = `
.c0 {
  background-color: transparent;
  border-color: transparent;
  box-sizing: border-box;
  font-family: Arial;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c0:focus {
  outline: none;
}

.c0::-moz-focus-inner {
  border: 0;
}

.c0 {
  background-color: #0000ff;
  color: #fff;
}

.c0:focus:not([disabled]),
.c0:hover:not([disabled]) {
  background-color: #fff;
  border-color: #0000ff;
  color: #0000ff;
}

.c0:active {
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
  -webkit-animation-name: click;
  animation-name: click;
  -webkit-animation-timing-function: ease-in;
  animation-timing-function: ease-in;
}

<div>
  <button
    class="c0"
    kind="primary"
  />
</div>
`;

exports[`Default button style grey button should match snapshot 1`] = `
.c0 {
  background-color: transparent;
  border-color: transparent;
  box-sizing: border-box;
  font-family: Arial;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c0:focus {
  outline: none;
}

.c0::-moz-focus-inner {
  border: 0;
}

.c0 {
  background-color: rgba(0,0,0,0.2);
}

.c0:focus:not([disabled]),
.c0:hover:not([disabled]) {
  background-color: #fff;
  border-color: rgba(0,0,0,0.2);
}

.c0:active {
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
  -webkit-animation-name: click;
  animation-name: click;
  -webkit-animation-timing-function: ease-in;
  animation-timing-function: ease-in;
}

<div>
  <button
    class="c0"
    kind="grey"
  />
</div>
`;

exports[`Default button style optional props should match snapshot 1`] = `
.c0 {
  background-color: transparent;
  border-color: transparent;
  box-sizing: border-box;
  font-family: Arial;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c0:focus {
  outline: none;
}

.c0::-moz-focus-inner {
  border: 0;
}

.c0 {
  cursor: not-allowed;
  opacity: 0.6;
}

.c0 {
  box-shadow: 0 2px 5px 0 rgba(0,0,0,0.26);
}

.c0 {
  background-color: #0000ff;
  color: #fff;
}

.c0:focus:not([disabled]),
.c0:hover:not([disabled]) {
  background-color: #fff;
  border-color: #0000ff;
  color: #0000ff;
}

.c0:active {
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
  -webkit-animation-name: click;
  animation-name: click;
  -webkit-animation-timing-function: ease-in;
  animation-timing-function: ease-in;
}

<div>
  <button
    class="c0"
    disabled=""
    kind="primary"
  />
</div>
`;

exports[`Default button style primary should match snapshot 1`] = `
.c0 {
  background-color: transparent;
  border-color: transparent;
  box-sizing: border-box;
  font-family: Arial;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c0:focus {
  outline: none;
}

.c0::-moz-focus-inner {
  border: 0;
}

.c0 {
  background-color: #0000ff;
  color: #fff;
}

.c0:focus:not([disabled]),
.c0:hover:not([disabled]) {
  background-color: #fff;
  border-color: #0000ff;
  color: #0000ff;
}

.c0:active {
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
  -webkit-animation-name: click;
  animation-name: click;
  -webkit-animation-timing-function: ease-in;
  animation-timing-function: ease-in;
}

<div>
  <button
    class="c0"
    kind="primary"
  />
</div>
`;

exports[`Default button style secondary should match snapshot 1`] = `
.c0 {
  background-color: transparent;
  border-color: transparent;
  box-sizing: border-box;
  font-family: Arial;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c0:focus {
  outline: none;
}

.c0::-moz-focus-inner {
  border: 0;
}

.c0 {
  background-color: #00ffff;
  color: #fff;
}

.c0:focus:not([disabled]),
.c0:hover:not([disabled]) {
  background-color: #fff;
  border-color: #00ffff;
  color: #00ffff;
}

.c0:active {
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
  -webkit-animation-name: click;
  animation-name: click;
  -webkit-animation-timing-function: ease-in;
  animation-timing-function: ease-in;
}

<div>
  <button
    class="c0"
    kind="secondary"
  />
</div>
`;

exports[`Default button style success should match snapshot 1`] = `
.c0 {
  background-color: transparent;
  border-color: transparent;
  box-sizing: border-box;
  font-family: Arial;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c0:focus {
  outline: none;
}

.c0::-moz-focus-inner {
  border: 0;
}

.c0 {
  background-color: #00ff00;
  color: #fff;
}

.c0:focus:not([disabled]),
.c0:hover:not([disabled]) {
  background-color: #fff;
  border-color: #00ff00;
  color: #00ff00;
}

.c0:active {
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
  -webkit-animation-name: click;
  animation-name: click;
  -webkit-animation-timing-function: ease-in;
  animation-timing-function: ease-in;
}

<div>
  <button
    class="c0"
    kind="success"
  />
</div>
`;

exports[`Default button style warning should match snapshot 1`] = `
.c0 {
  background-color: transparent;
  border-color: transparent;
  box-sizing: border-box;
  font-family: Arial;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c0:focus {
  outline: none;
}

.c0::-moz-focus-inner {
  border: 0;
}

.c0 {
  background-color: #ff0000;
  color: #fff;
}

.c0:focus:not([disabled]),
.c0:hover:not([disabled]) {
  background-color: #fff;
  border-color: #ff0000;
  color: #ff0000;
}

.c0:active {
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
  -webkit-animation-name: click;
  animation-name: click;
  -webkit-animation-timing-function: ease-in;
  animation-timing-function: ease-in;
}

<div>
  <button
    class="c0"
    kind="warning"
  />
</div>
`;
